File: example/example_1.py
"""例題1：ファイル操作
example_1_dataフォルダにあるcsvファイルの中で，変数xにある数値以上の値が含まれる物をリストアップするプログラムを作成してください．
結果はcsvファイルでexample_1_rezultフォルダに保存されるようにしてください．
ただし結果のファイル名には実行時刻が入るようにしてください．
"""

### Pandasを使用 ###
import pandas as pd
import pathlib
import datetime
import csv

def func_example_1(path_data, path_result, threshold):
    """処理をすべて行う関数"""
    
    date_now = datetime.datetime.now()  # 日付取得．保存ファイル名用
    csv_names = path_data.glob('*.csv')  # csvファイルの名前を取得
    header = 'Files with x of ' + str(threshold) + ' or more is'
    result = [[header]]  # 結果格納
    
    for n in csv_names:
        file_name = n.name
        path_csv = path_data / file_name
        df = pd.read_csv(path_csv)  # csvファイル読み込み
        df = df.fillna(0)  # 欠損値を0で置き換え
        if (df['x'] >= 40).any():
            result.append([file_name])
        else:
            pass
    
    # 結果をcsvファイルで出力
    result_name = 'rezult_' + date_now.strftime('%Y-%m-%d--%H-%M-%S') + '.csv'
    path_result = path_result / result_name
    with open(path_result, 'x', newline="") as f:
        writer = csv.writer(f, delimiter=',')
        writer.writerows(result)
    
    return None


if __name__ == '__main__':
    re_path_data = pathlib.Path('example_1_data')  # exercize_1_dataフォルダの相対パス
    ab_path_data = re_path_data.resolve()  # 絶対パスに変換
    
    re_path_result = pathlib.Path('example_1_result')  # exercize_1_rezultフォルダの相対パス
    ab_path_result = re_path_result.resolve()  # 絶対パスに変換
    
    func_example_1(
        path_data = ab_path_data,
        path_result = ab_path_result,
        threshold = 40,
    )

File: example/example_1.m
function example_1()
% 例題1：ファイル操作
%   example_1_dataフォルダにあるcsvファイルの中で，変数xにある数値以上の値が...
%   含まれる物をリストアップするプログラムを作成してください．
%   結果はcsvファイルでexample_1_rezultフォルダに保存されるようにしてください．
%   ただし結果のファイル名には実行時刻が入るようにしてください．

%% 日時取得
dt = datetime('now');
DateString = datestr(dt,'yyyy-MM-dd-HH-mm-ss-FFF');

%% ファイル，フォルダのパス取得
currentFolder = pwd;  % 現在のフォルダの絶対パス
dataFolder = append(currentFolder, '/example_1_data');  % csvの保存先フォルダ
outputFolder = append(currentFolder, '/example_1_result');  % 結果の保存先フォルダ
cd example_1_data  % フォルダへ移動
csv_list = dir(append(dataFolder, '/*.csv'));  % ワイルドカードで~.csvの名前取得
cd ..  % 元のフォルダに戻る

%% 処理
threshold = 40;  % 閾値
output = [];  % 条件を満たすcavファイル名を放り込む配列．

for i = 1:1:length(csv_list)  % ファイルの数だけループ
    temp_table = readtable([csv_list(i).folder '/' csv_list(i).name]);  % csvをtableとして読み込み
    temp_table = rmmissing(temp_table);  % 欠損値削除
    
    if any(temp_table.x > threshold)  % 閾値判定
        output = [output; string(csv_list(i).name)];  % 結果の配列に入れる
    else
        % 何もしない
    end    
end

%% 結果をcsvファイル出力
output_name = append(DateString, '_output.csv');
output_path = append(outputFolder, '/', output_name);
writematrix(output, output_path)

end